Chapter08/recipe-05/cxx-example/cmake/Testing.cmake

```diff
@@ -0,0 +1,58 @@
+include(CMakeParseArguments)
+
+macro(add_catch_test)
+  set(options)
+  set(oneValueArgs NAME COST)
+  set(multiValueArgs LABELS DEPENDS REFERENCE_FILES)
+  cmake_parse_arguments(add_catch_test
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN}
+    )
+
+  message(STATUS "defining a test ...")
+  message(STATUS "    NAME: ${add_catch_test_NAME}")
+  message(STATUS "    LABELS: ${add_catch_test_LABELS}")
+  message(STATUS "    COST: ${add_catch_test_COST}")
+  message(STATUS "    REFERENCE_FILES: ${add_catch_test_REFERENCE_FILES}")
+
+  add_test(
+    NAME
+      ${add_catch_test_NAME}
+    COMMAND
```

Comment:
Now that I think about it, maybe also the command should become an argument to `add_Catch_test`?

Reply:
OK I will adapt it. I reused a lot of your recipe for recipe 1 but testing was the only thing I could think of where one needs to repeatedly call a function/macro and this is what I wanted to demo.

Reply:
Sure, that's a great example for refactoring CMake code too!

Reply:
I looked at it a bit more and think that we should leave it as is but discuss this possibility only in the text. The additional generality will lead to more text that we need to explain and fit into a limited line.

Reply:
I have added a place-holder in the book so I don't forget to mention this.